### PR TITLE
[CI] Update docker images tag to 20240105-165030-51bdaec6

### PR DESCRIPTION
**Pull request by vinx13**
This is to integrate changes in #16336

### CODE REVIEW

**Review by junrushao — APPROVED** [review]

**Comment by junrushao** [comment]
not sure about the CI failures...

**Comment by vinx13** [comment]
[this](https://forums.developer.nvidia.com/t/unable-to-start-cuda-container-with-recent-update-on-november-10/272883) seems related to ci gpu error. CI has cuda 11.7 installed while the docker image has 11.8, which can be run until the Nvidia updates on base image on November. We need to update the AMIs for GPU workers. @tqchen

**Review by yongwww — APPROVED** [review]
lgtm

File: ci/jenkins/generated/gpu_jenkinsfile.groovy [hunk 1]
```diff
@@ -112,7 +112,7 @@ properties([
 upstream_revision = null
 
 // command to start a docker container
-docker_run = 'docker/bash.sh --env CI --env PLATFORM --env TVM_SHARD_INDEX --env TVM_NUM_SHARDS --env RUN_DISPLAY_URL --env PLATFORM --env SKIP_SLOW_TESTS --env TEST_STEP_NAME'
+docker_run = 'docker/bash.sh --env CI --env PLATFORM --env TVM_SHARD_INDEX --env TVM_NUM_SHARDS --env RUN_DISPLAY_URL --env PLATFORM --env SKIP_SLOW_TESTS --env TEST_STEP_NAME --env NVIDIA_DISABLE_REQUIRE=true'
```
Comment (yongwww) [inline ci/jenkins/generated/gpu_jenkinsfile.groovy:115]: Thanks for this effort! doc about the [NVIDIA_DISABLE_REQUIRE  variable](https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/docker-specialized.html#nvidia-disable-require-environment-variable). We may consider disabling it when upgrading the image/cuda in the future.

**Pull request merged**